### PR TITLE
tree-wide: cargo fmt

### PR DESCRIPTION
**Pull request by cgwalters**
And add this to CI.  One good thing about Rust and Go (among others) is that there's a "standard" formatter; my best guess here is this repository isn't formatted in the default way just because you aren't using a development environment which encourages it.  It's easy to keep things formatted with a simple `cargo fmt`; do so and enforce it in CI.

### CODE REVIEW

File: src/handlers/ar.rs [hunk 1]
```diff
@@ -141,9 +175,9 @@ mod tests {
         let cfg = Rc::new(options::Config::empty(0));
         let h = Ar::boxed(&cfg);
 
-        assert!( h.filter(Path::new("/some/path/libfoobar.a")).unwrap());
+        assert!(h.filter(Path::new("/some/path/libfoobar.a")).unwrap());
         assert!(!h.filter(Path::new("/some/path/libfoobar.aa")).unwrap());
-        assert!( h.filter(Path::new("/some/path/libfoobar.a.a")).unwrap());
+        assert!(h.filter(Path::new("/some/path/libfoobar.a.a")).unwrap());
```
Comment (keszybz) [inline src/handlers/ar.rs:180]: Not better.

File: src/handlers/ar.rs [hunk 2]
```diff
@@ -99,11 +122,22 @@ impl super::Processor for Ar {
...
+                debug!(
+                    "{}: file {:?}, mtime={}, {}:{}, mode={:o}, size={}",
+                    io.input_path.display(),
+                    name,
+                    mtime,
+                    uid,
+                    gid,
+                    mode,
+                    size
+                );
```
Comment (keszybz) [inline src/handlers/ar.rs:134]: Not better.

File: src/handlers/ar.rs [hunk 3]
```diff
@@ -74,8 +88,13 @@ impl super::Processor for Ar {
...
             if &buf[58..] != FILE_HEADER_MAGIC {
-                return Err(anyhow!("{}: wrong magic in file header at offset {}: {:?} != {:?}",
-                                   io.input_path.display(), pos, &buf[58..], FILE_HEADER_MAGIC));
+                return Err(anyhow!(
+                    "{}: wrong magic in file header at offset {}: {:?} != {:?}",
+                    io.input_path.display(),
+                    pos,
+                    &buf[58..],
+                    FILE_HEADER_MAGIC
+                ));
```
Comment (keszybz) [inline src/handlers/ar.rs:97]: Not better.

File: src/handlers/mod.rs [hunk 4]
```diff
@@ -35,24 +35,21 @@ pub trait Processor {
...
     ("javadoc", javadoc::Javadoc::boxed),
-    ("pyc",     pyc::Pyc::boxed),
+    ("pyc", pyc::Pyc::boxed),
 ];
 
 pub fn handler_names() -> Vec<&'static str> {
-    HANDLERS.iter()
-        .map(|(name, _)| *name)
-        .collect()
+    HANDLERS.iter().map(|(name, _)| *name).collect()
```
Comment (keszybz) [inline src/handlers/mod.rs:52]: Not better.

File: src/handlers/pyc.rs [hunk 5]
```diff
@@ -530,13 +547,13 @@ impl PycParser {
...
         let offset = self.take(4)?;
-        let bytes = &self.data[offset .. offset + 4];
+        let bytes = &self.data[offset..offset + 4];
         Ok(u32::from_le_bytes(bytes.try_into().unwrap()))
     }
 
     fn _read_long_signed(&mut self) -> Result<i32> {
         let offset = self.take(4)?;
-        let bytes = &self.data[offset .. offset + 4];
+        let bytes = &self.data[offset..offset + 4];
```
Comment (keszybz) [inline src/handlers/pyc.rs:556]: That's not easier to read…

**Review by keszybz — CHANGES_REQUESTED** [review]
I don't want to do this. I know people like automatic formatting, but in many cases it is significantly worse than manual formatting. In particular, when there are repeated cases with similar structure, consistency between them is more important than adhering to some arbitrary length of lines. I also find CI enforcement of formatting very obnoxious.

Also, debug statements are an annoyance to be suffered, so they should be fairly concise. Making them spread to 10 lines each makes the code harder to read.

Though many of the changes here are good. E.g. I see I was adding trailing commas in `match` where they are not necessary.

File: src/handlers/ar.rs (hunk 2 above)
Comment (cgwalters) [reply src/handlers/ar.rs:134]: Yes, I agree.  Though the real fix here is https://github.com/keszybz/add-determinism/pull/15 I think.

**Comment by cgwalters** [comment]
Yeah, I understand. I just personally acclimated to it, though I may at some point try to bump up the default line length to something higher like 110 at least across my projects. Bear in mind that "hard rustfmt" usage is pretty widespread IME in the community. It may be worth trying to play with the rustfmt configuration options to see if it can get closer to what you want, as it really does make it *much* easier for contributors IMO to not have to worry about style matching.

**Comment by keszybz** [comment]
I extracted half of the patch and pushed it to main.

**Comment by cgwalters** [comment]
There's also e.g. `#[rustfmt::skip]`: https://github.com/rust-lang/rustfmt?tab=readme-ov-file#tips

**Pull request closed without merging**